ci/test-rest-args.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1091,SC1090
+
+# test that we correctly handle args after `--`
+
+set -x
+set -eo pipefail
+
+ci_dir=$(dirname "${BASH_SOURCE[0]}")
+ci_dir=$(realpath "${ci_dir}")
+. "${ci_dir}"/shared.sh
+
+main() {
```

Comment:
This needs a `clippy -- --dent warnings` just to ensure that `-Zbuild-std` is getting correctly added as well.